functions/Move-DbaDbFile.ps1

```diff
@@ -0,0 +1,438 @@
+                }
+            } else {
+                if ($fileTypeFilter -eq -1) {
+                    $DataFiles = Get-DbaDbFile -SqlInstance $server -Database $Database | Select-Object LogicalName, PhysicalName
+                } else {
+                    $DataFiles = Get-DbaDbFile -SqlInstance $server -Database $Database | Where-Object Type -eq $fileTypeFilter | Select-Object LogicalName, PhysicalName
+                }
+            }
+
+            if (@($DataFiles).Count -gt 0) {
+
+                if ($FileStructureOnly) {
+                    $fileStructure = "`$fileToMove=@{`n"
+                    foreach ($file in $DataFiles) {
+                        $fileStructure += "`t'$($file.LogicalName)'='$(Split-Path -Path $file.PhysicalName -Parent)'`n"
+                    }
+                    $fileStructure += "}"
+                    Write-Output $fileStructure
+                    return
+                }
+
+                if ($FileDestination) {
+                    $DataFilesToMove = $DataFiles | Select-Object -ExpandProperty LogicalName
+                } else {
+                    $DataFilesToMove = $FileToMove.Keys
+                }
+
+                if ($dbStatus -ne "Offline") {
+                    if ($PSCmdlet.ShouldProcess($database, "Setting database $Database offline")) {
+                        try {
+
+                            $SetState = Set-DbaDbState -SqlInstance $server -Database $Database -Offline -Force:$Force
+                            if ($SetState.Status -ne 'Offline') {
+                                Write-Message -Level Warning -Message "Setting database Offline failed!"
+
+                            } else {
+                                Write-Message -Level Verbose -Message "Database $Database was set to Offline status."
+                            }
+                        } catch {
+                            Stop-Function -Message "Setting database Offline failed! : $($_.Exception.InnerException.InnerException.InnerException)" -ErrorRecord $_ -Target $server.DomainInstanceName -OverrideExceptionMessage
+                        }
+                    }
+                }
+
+                $locally = $false
+                if ([DbaValidate]::IsLocalhost($server.ComputerName)) {
+                    # locally ran so we can just use Start-BitsTransfer
+                    $ComputerName = $server.ComputerName
+                    $locally = $true
+                } else {
+                    # let's start checking if we can access .ComputerName
+                    $testPS = $false
+                    if ($SqlCredential) {
+                        # why does Test-PSRemoting require a Credential param ? this is ugly...
+                        $testPS = Test-PSRemoting -ComputerName $server.ComputerName -Credential $SqlCredential -ErrorAction Stop
```

Comment:
This should pass `$Credential`

Reply:
The comment can be removed as well. The parameter is required because utilizing WSMan requires elevated rights by default on a Windows Server.